Opencast-Moodle/moodle-block_opencast
# LTI modules cleanup process scheduled task

This PR fixes #188 

### Description
refer to [issue description](https://github.com/Opencast-Moodle/moodle-block_opencast/issues/188#issue-878858834)

### Solution
A new scheduled task (cron job)  `\block_opencast\task\cleanup_lti_module_cron` is introduced to provide the LTI modules cleanup process, in which at first the manually added lti modules are recorded and after that is conducts an in-depth validation check on every single entries of both ltiepisode and ltimodule tables in moodle.


### How to test this PR:
- make sure the admin setting for add lti modules are configured properly.
- make sure an upgrade of this plugin is in place.
- make sure scheduled task (cronjobs) of moodle is enabled and working.
- make sure that `block_opencast\task\cleanup_lti_module_cron` scheduled task runs every minute.
- make sure the moodle system has at least one LTI preconfigured tool for each series and episode separately.
- try to add both episode lti module and series lti module in a course.
- run the scheduled task and check the db if the newlly added records are also recorded in `mdl_block_opencast_ltiepisode` and `mdl_block_opencast_ltimodule`
- try to either change and make existing lti modules faulty, or add a faulty new one manually, and check if the record exists in the db (above mentioned tables)
PS: Please refer to the scheduled taks log in admin setting to check the messages
